### ISSUE TITLE
lxcfs gives incorrect memory usage

### ISSUE THREAD

**Issue by pikrzysztof** [lxc/lxcfs#265]
host machine:
```
kris@hex-64b-pm:~$ lxcfs --version
3.0.2
kris@hex-64b-pm:~$ uname -a
Linux hex-64b-pm 4.18.0-11-generic #12-Ubuntu SMP Tue Oct 23 19:22:37 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
kris@hex-64b-pm:~$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.1 LTS"
VERSION_ID="18.04"
<snip>
UBUNTU_CODENAME=bionic
```
LXC:
```
kris@1804-tmpl:/proc$ cat /proc/meminfo 
MemTotal:       33554432 kB
MemFree:        31191452 kB
MemAvailable:   31322264 kB
Buffers:               0 kB
Cached:           130812 kB
SwapCached:            0 kB
Active:           109744 kB
Inactive:          64072 kB
Active(anon):      41528 kB
Inactive(anon):      104 kB
Active(file):      68216 kB
Inactive(file):    63968 kB
Unevictable:           0 kB
Mlocked:               0 kB
SwapTotal:             0 kB
SwapFree:              0 kB
Dirty:               616 kB
Writeback:             0 kB
AnonPages:       2548768 kB
Mapped:           287948 kB
Shmem:               108 kB
Slab:               0 kB
SReclaimable:          0 kB
SUnreclaim:            0 kB
KernelStack:        8776 kB
PageTables:        17016 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:    32970524 kB
Committed_AS:    6850260 kB
VmallocTotal:   34359738367 kB
VmallocUsed:           0 kB
VmallocChunk:          0 kB
HardwareCorrupted:     0 kB
AnonHugePages:         0 kB
ShmemHugePages:        0 kB
ShmemPmdMapped:        0 kB
CmaTotal:              0 kB
CmaFree:               0 kB
HugePages_Total:       0
HugePages_Free:        0
HugePages_Rsvd:        0
HugePages_Surp:        0
Hugepagesize:       2048 kB
Hugetlb:               0 kB
DirectMap4k:     6956624 kB
DirectMap2M:    28665856 kB
DirectMap1G:    33554432 kB
kris@1804-tmpl:/sys/fs/cgroup/memory$ cat memory.stat 
cache 122433536
rss 38023168
rss_huge 0
shmem 110592
mapped_file 51363840
dirty 135168
writeback 0
swap 0
pgpgin 20128317
pgpgout 20089042
pgfault 35105169
pgmajfault 29898
inactive_anon 106496
active_anon 38043648
inactive_file 63582208
active_file 59408384
unevictable 0
hierarchical_memory_limit 34359738368
hierarchical_memsw_limit 34359738368
total_cache 133951488
total_rss 43003904
total_rss_huge 0
total_shmem 110592
total_mapped_file 52580352
total_dirty 135168
total_writeback 0
total_swap 0
total_pgpgin 21784323
total_pgpgout 21741122
total_pgfault 37648611
total_pgmajfault 33660
total_inactive_anon 106496
total_active_anon 42442752
total_inactive_file 65503232
total_active_file 69853184
total_unevictable 0
kris@1804-tmpl:/sys/fs/cgroup/memory$ cat memory.usage_in_bytes 
2419781632
kris@1804-tmpl:/sys/fs/cgroup/memory$ free
              total        used        free      shared  buff/cache   available
Mem:       33554432     2232176    31191444         108      130812    31322256
Swap:             0           0           0
```

`top`: 
![screenshot from 2019-01-04 17-42-14](https://user-images.githubusercontent.com/3964215/50702210-21aa2800-1048-11e9-9ce5-e01e47350cbe.png)

From [kernel cgroups docs](https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt):

> 5.5 usage_in_bytes
> 
> For efficiency, as other kernel components, memory cgroup uses some optimization
to avoid unnecessary cacheline false sharing. usage_in_bytes is affected by the
method and doesn't show 'exact' value of memory (and swap) usage, it's a fuzz
value for efficient access. (Of course, when necessary, it's synchronized.)
If you want to know more exact memory usage, you should use RSS+CACHE(+SWAP)
value in memory.stat(see 5.2).

In conclusion: I'm using ~100MiB memory in the container. The container reports that I'm using 2GiB.

**Comment by stgraber** [comment]
Any chance you can test with a current build of lxcfs?
